tests/phpunit/Unit/iCalendar/DateParserTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace SRF\Tests\iCalendar;
+
+use SRF\iCalendar\DateParser;
+
+/**
+ * @covers \SRF\iCalendar\DateParser
+ * @group semantic-result-formats
+ *
+ * @license GNU GPL v2+
+ * @since 3.1
+ *
+ * @author mwjames
+ */
+class DateParserTest extends \PHPUnit_Framework_TestCase {
+
+	public function testCanConstruct() {
+
+		$this->assertInstanceOf(
+			DateParser::class,
+			new DateParser()
+		);
+	}
+
+	public function testParseDate_Year() {
+
+		$timeValue = $this->getMockBuilder( '\SMWTimeValue' )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$timeValue->expects( $this->any() )
```

Comment:
SMWTimeValue is a value object right? Then why are you creating a mock? This adds a bunch of hidden coupling (method names as strings, plus their signatures...) which is best avoided if there is no good reason for it.

Reply:
> SMWTimeValue is a value object right? 

Sort of but using an a mock ensures that specific coupling to let's say a Title object, or the Store are not pulled in hereby avoids creating a possible DB connection, especially the Title object is nasty, and by using a mock I'm certain I have an encapsulate object that doesn't create any connection.

Reply:
Alright, sounds like SMWTimeValue is not a value object then.